makefile

```diff
@@ -0,0 +1,85 @@
+# Makefile for shipping and testing the container image.
+
+MAKEFLAGS += --warn-undefined-variables
+.DEFAULT_GOAL := build
+.PHONY: *
+
+# we get these from CI environment if available, otherwise from git
+GIT_COMMIT ?= $(shell git rev-parse --short HEAD)
+GIT_BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+
+namespace ?= autopilotpattern
+tag := branch-$(shell basename $(GIT_BRANCH))
+image := $(namespace)/nats
+test_image := $(namespace)/nats-testrunner
+
+dockerLocal := DOCKER_HOST= DOCKER_TLS_VERIFY= DOCKER_CERT_PATH= docker
+
+## Display this help message
+help:
+	@awk '/^##.*$$/,/[a-zA-Z_-]+:/' $(MAKEFILE_LIST) | awk '!(NR%2){print $$0p}{p=$$0}' | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' | sort
+
+
+# ------------------------------------------------
+# Container builds
+
+## Builds the application container image locally
+build: test-runner
+	$(dockerLocal) build -t=$(image):$(tag) .
+
+## Build the test running container
+test-runner:
+	$(dockerLocal) build -f test/Dockerfile -t=$(test_image):$(tag) .
+
+## Push the current application container images to the Docker Hub
+push:
+	$(dockerLocal) push $(image):$(tag)
+
+## Tag the current images as 'latest' and push them to the Docker Hub
+ship:
+	$(dockerLocal) tag $(image):$(tag) $(image):latest
+	$(dockerLocal) push $(image):$(tag)
+	$(dockerLocal) push $(image):latest
+
+
+# ------------------------------------------------
+# Test running
+
+## Pull the container images from the Docker Hub
+pull:
+	docker pull $(image):$(tag)
+
+$(DOCKER_CERT_PATH)/key.pub:
+	ssh-keygen -y -f $(DOCKER_CERT_PATH)/key.pem > $(DOCKER_CERT_PATH)/key.pub
```

Comment:
You should try setting up the test job on Jenkins... I'm fairly certain we don't need this here.